core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/timers/TimerUtils.java

```diff
@@ -59,4 +63,33 @@ private TimerUtils() {
 
         return timeoutTracker;
     }
+
+    /**
+     * Schedule a {@link TimeoutTask} that aborts the task if not otherwise completed before the given timeout.
+     *
+     * @param timeoutExecutor the executor to execute the {@link TimeoutTask}
+     * @return a {@link TimeoutTracker}
```

Comment:
Please add docs for the other params as well. It's hard to tell what `isInterruptedThread` is without looking at the source. I'd also suggest changing it to `interruptCurrentThread`